### PR TITLE
Add patch to prevent NPE of pos

### PR DESCRIPTION
**Pull request by liach**
With a modded way to sleep, e.x. a minecart that allows sleeping, the bedPosition may be null and cause a npe in block pos boundary check.

Signed-off-by: liach <liach@users.noreply.github.com>

Edit: I really wish people can review this pull so that I get this *crash* fixed. It is a crash.

Edit 2: Why would I fix the crash instead of setting a position? Because that position is used to reset spawn if it is not set to null.

### CODE REVIEW

**Comment by liach** [comment]
It's very weird that I discovered an issue with entity spawning with incorrect factory on the client side after getting an entity message packet. I will make a change for that too so that my test mod works correctly.

File: src/test/java/net/minecraftforge/debug/entity/player/PlayerSleepWithoutPosTest.java [hunk 1]
```diff
@@ -0,0 +1,340 @@
...
+import static net.minecraftforge.debug.entity.player.PlayerSleepWithoutPosTest.*;
+
+/**
+ * A test mod that adds a minecart in which the player can sleep. The minecart can only be summoned by commands.
+ */
+@SuppressWarnings("unused")
+@Mod(modid = MOD_ID, name = "Player Sleep Without Position Test", acceptableRemoteVersions = "*")
+public final class PlayerSleepWithoutPosTest
+{
+    private static final boolean ENABLED = true;
```
Comment (liach) [inline src/test/java/net/minecraftforge/debug/entity/player/PlayerSleepWithoutPosTest.java]: I will edit this field later.

File: src/test/resources/assets/player_sleep_without_pos_test/lang/en_US.lang [hunk 2]
```diff
@@ -0,0 +1,3 @@
+player_sleep_without_pos_test.key.sleep=Minecart Sleep Key
```
Comment (liach) [inline src/test/resources/assets/player_sleep_without_pos_test/lang/en_US.lang]: I will edit the file name later.

**Comment by HenryLoenwind** [comment]
You may want to remove the `cls = er.getEntityClass()`, too. The newInstance() is the only consumer.

And go though each use of `net.minecraftforge.fml.common.registry.EntityRegistry.EntityRegistration.entityClass` to see if that's wrong, too. (only `net.minecraftforge.fml.common.registry.EntityRegistry.addSpawn()` and `removeSpawn()` as far as I can see.)

**Comment by bs2609** [comment]
Related: #4845, #4846

**Comment by liach** [comment]
@HenryLoenwind that was in fact the only place where that class was used.

For both henry and @bs2609 , would you comment on my approach of entity registration changes?

**Comment by HenryLoenwind** [comment]
I'm no expert on entities, but that patch looks good to me.

You may want to split off your original NPE patch from this to get that one merged sooner. ;)

**Comment by liach** [comment]
uhh, i opened my pr for the npe patch and now rediscovered the old bug.

**Pull request closed without merging**